test/jdk/valhalla/valuetypes/StaticFactoryMethodHandleTest.java

```diff
@@ -64,34 +64,34 @@ private ConstructorWithArgs(int x, int y, int z) {
 
     public static void main(String... args) throws Throwable {
         // test default static init factory
-        MethodType mtype0 = MethodType.methodType(DefaultConstructor.class);
-        MethodHandle mh0 = staticInitFactory(DefaultConstructor.class, mtype0);
+        MethodType mtype0 = MethodType.methodType(DefaultConstructor.class.asValueType());
+        MethodHandle mh0 = staticInitFactory(DefaultConstructor.val.class, mtype0);
         DefaultConstructor o0 = (DefaultConstructor)mh0.invokeExact();
```

Comment:
While the preferred change is go from Point.class to Point.class.asValueType(), here is one of the few places I had to resort to Point.val.class - This is because we don't have a spec for Class#asValueType() similar to Object#getClass() where the compiler munges the return type of the method. As a result, code won't compile unless I change it to Point.val.class format

Reply:
What special treatment of `Object#getClass` in javac?

It's fine with this temporary workaround.  FYI.  The discussion about disallowing`Point.val.class` and `Point.ref.class` syntax to obtain the primitive value type mirror but use the `Class` API instead is relevant here.